### PR TITLE
docs: Add commit messages styleguide

### LINKED ISSUE #45: Standardization of git commit messages
A git repository lives with an history that let developers or automatic procedure to find useful information. Therefore a minimum of standardization is very valuable to every steps in a development/maintenance workflow.

Commits' messages are a part of development and commits' messages guidelines are part of development guidelines.

I found a very welldone commits's messages styleguide: https://udacity.github.io/git-styleguide/

If maintainers agree, I will add a Pull-Request to enhance the current styleguide with some kind of commits' messages styleguide inspired by udacity's one.

### PR DESCRIPTION
**Pull request by enguerran**
strongly inspired by https://udacity.github.io/git-styleguide/

fix #45

### CODE REVIEW

**Comment by gregorylegarec** [comment]
Thanks, it looks nice. I'd just love to add something about emoji prefixes, because I love them so much.

File: README.md [hunk 1]
```diff
@@ -186,6 +186,135 @@ const App = props => (
...
+    <summary>See more</summary>
+<p>
+
+##### Type
+
+One of:
+
+- __feat__: a new feature
+- __fix__: a bug fix
+- __docs__: changes to documentation
```
Comment (kosssi) [inline README.md]: Pourquoi mettre un `s` à `docs` et pas à `style` ou `test` ?

Comment (gregorylegarec) [reply README.md]: Je pense que c'est pour être raccord avec commitizen

![https://raw.githubusercontent.com/commitizen/cz-cli/master/meta/screenshots/add-commit.png]

Comment (drazik) [reply README.md]: That's just what [conventional commits](https://conventionalcommits.org/) says : 

> Others: commit types other than fix: and feat: are allowed, for example commitlint-config-conventional (based on the the Angular convention) recommends chore:, docs:, style:, refactor:, perf:, test:, and others.

**Comment by gregorylegarec** [comment]
@enguerran Visual value. Use them as prefix permit to see them aligned vertically, and never hidden in case of a too long commit message.

File: README.md (hunk 1 above)
Comment (enguerran) [reply README.md]: I don't have an answer as I copied/pasted what udacity does.
But if you want some justification, see https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch that speaks about `/docs`. Every single other prefix are singular.

**Comment by ptbrowne** [comment]
:shipit:

**Comment by enguerran** [comment]
Pros/Cons of using emojis in the prefix of a git commit message's subject.

__Pros:__
1. Visual value: using them as prefix permit to see them aligned vertically

__Cons:__
1. Not very easy to add emoji in some users' environment
1. Duplicate of the prefix word
1. Not used by others git commit messages conventions/stylguides: [angular](https://gist.github.com/stephenparish/9941e89d80e2bc58a153), [commit.style](https://commit.style/), [Chris Beams](https://chris.beams.io/posts/git-commit/), [udacity](https://udacity.github.io/git-styleguide/)

Could you help me to fill this comparaison?

---

> and never hidden in case of a too long commit message.

Note that a commit message is limited in size: _« Subjects should be no greater than 50 characters »_

**Comment by ptbrowne** [comment]
I agree with Enguerran, the fact that we had to write custom code to support emojis does not push me to add them. I also had to configure a custom pager for my git for it to properly dispaly them. I love emojis as much as the next guy, but I personally think it has no place in the official guidelines.

**Comment by gregorylegarec** [comment]
@ptbrowne You're right, there is no real need to put them into our guidelines.

**Comment by gregorylegarec** [comment]
Thanks @enguerran 👌

File: README.md (hunk 1 above)
Comment (kosssi) [reply README.md]: I ask directly here: https://github.com/conventional-changelog/conventionalcommits.org/issues/61

**Pull request merged**